iroh-net/src/magicsock/derp_actor.rs

```diff
@@ -373,12 +458,25 @@ impl DerpActor {
         let now = Instant::now();
 
         let mut to_close = Vec::new();
-        for (i, ad) in &self.active_derp {
+        for (i, (s, _)) in &self.active_derp {
             if *i == self.conn.my_derp() {
                 continue;
             }
-            if ad.last_write.duration_since(now) > DERP_INACTIVE_CLEANUP_TIME {
-                to_close.push(*i);
+            let (os, or) = oneshot::channel();
+            match s.send(ActiveDerpMessage::GetLastWrite(os)).await {
+                Ok(_) => match or.await {
+                    Ok(last_write) => {
+                        if last_write.duration_since(now) > DERP_INACTIVE_CLEANUP_TIME {
```

Comment:
Just trying to remember, is the idea with `clean_stale_derp` and other methods that close the derp clients, that if we end up needing to dial that derper later, we can just add it back as an `active_derp` when we need it?

Reply:
yes